verifier-py/tests-unit/test_playbook_verifier.py

```diff
@@ -0,0 +1,261 @@
+
+    def test_from_env(self, monkeypatch):
+        temp = tempfile.NamedTemporaryFile(prefix="insights-ansible-playbook-verifier-test-")
+        monkeypatch.setenv("INSIGHTS_ANSIBLE_PLAYBOOK_VERIFIER_GPG_KEY", temp.name)
+        tempkey = pathlib.Path(temp.name)
+        with tempkey.open("wb") as f:
+            f.write(b"-----BEGIN PGP PUBLIC KEY BLOCK-----\n")
+            f.write(b"this is example content\n")
+            f.write(b"-----END PGP PUBLIC KEY BLOCK-----\n")
+
+        key: bytes = verifier.get_gpg_key_content()
+
+        assert b"-----BEGIN PGP PUBLIC KEY BLOCK-----" in key
+        assert b"this is example content" in key
+        assert b"-----END PGP PUBLIC KEY BLOCK-----" in key
+
+
+class TestParsePlaybook:
+    """The reference verified used YAML 1.2.
+
+    PyYAML seems to be using YAML 1.1 by default, so we have to ensure we parse it correctly.
+    """
+
+    def test_first(self):
+        """Test that only the first item out of the list is loaded if multiple are present."""
+        raw = "\n".join(
+            [
+                "---",
+                "- name: first dictionary",
+                "  key: value",
+                "- name: second dictionary",
+                "  key: value",
+            ]
+        )
+        expected = {"name": "first dictionary", "key": "value"}
+
+        actual = verifier.parse_playbook(raw)
+
+        assert actual == expected
+
+    def test_integers(self):
+        raw = "- [1, 2, 3]"
+        expected = [1, 2, 3]
+
+        actual = verifier.parse_playbook(raw)
+
+        assert actual == expected
+
+    def test_floats(self):
+        raw = "- [1.0, 2.0, 3.0]"
+        expected = [1.0, 2.0, 3.0]
+
+        actual = verifier.parse_playbook(raw)
+
+        assert actual == expected
+
+    def test_true(self):
+        raw = "- bool: [true, True, TRUE]\n  string: [y, yes, Yes, YES, on, On, ON]"
+        expected = {
+            "bool": [True, True, True],
+            "string": ["y", "yes", "Yes", "YES", "on", "On", "ON"],
+        }
+
+        actual = verifier.parse_playbook(raw)
+
+        assert actual == expected
+
+
+class TestCleanPlaybook:
+    def test_ok(self):
+        raw = {
+            "name": "good playbook",
+            "hosts": "localhost",
+            "vars": {
+                "insights_signature_exclude": "/hosts,/vars/insights_signature/",
+                "insights_signature": b"data",
+            },
+            "tasks": [],
+        }
+        expected = {
+            "name": "good playbook",
+            "vars": {"insights_signature_exclude": "/hosts,/vars/insights_signature/"},
+            "tasks": [],
+        }
+
+        actual: dict = verifier.clean_playbook(raw)
+        assert actual == expected
+
+    def test_requires_vars(self):
+        raw = {"name": "bad playbook", "tasks": [{"name": "a task"}]}
+
+        with pytest.raises(Exception, match="does not have the key 'vars"):
```

Comment:
this and all the other `pytest.raises(Exception, ...)` in this file: maybe better match on the actual exception, which is `PreconditionError`